docs/contributor-guide/developer-guide/technical-contribution-guide.md

```diff
@@ -18,9 +18,58 @@ Development tasks are managed in the [GitHub issues](https://github.com/civiform
 
 If you're just getting started, check out issues labeled with [Good First Issue](https://github.com/civiform/civiform/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22). Also check out issues in the [next milestone](https://github.com/civiform/civiform/milestones?direction=asc\&sort=due\_date\&state=open) so you can work on the highest-priority tasks.
 
+### Git workflow
+
+Get the latest state from the github repository and create a new local development branch:
+
+```sh
+git fetch origin
+git checkout -b your_username/dev-branch-description
+git reset origin/main
+```
+
+Then start working on your change.
+
+When you're ready to submit your code for review, package your changes into a commit:
+
+```sh
+git status # make sure you are only adding the expected files.
+git add .
```

Comment:
`git add .` / `git add -A` / `git add *` aren't great, since it can include extra files that you don't intend to push.  Better is to do either `git add -p` (for interactive mode) or `git status` and `git add <filename>` with the files you want to add